src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  ChangeEvent,
+  useContext,
+  useEffect,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { DispatchContext, StateContext } from '../../State/State';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  setIsEditing?: (v: boolean) => void
+  isLoading?: boolean,
+  setIsLoading?: (v: boolean) => void
+  currentTitle?: string,
+  setCurrentTitle?: (v: string) => void
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setIsEditing,
+  isLoading,
+  setIsLoading,
+  currentTitle,
+  setCurrentTitle,
+}) => {
+  const { title, completed, id } = todo;
+
+  const { clearAll, isEscapeKeyup } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  useEffect(() => {
+    if (clearAll && completed && setIsLoading) {
```

Comment:
what about `toggleAll` function - there also should be loading during `toggleAll`

I'd recommend you to save the array of IDs you want to show loading for
For example, if you delete 1 todo set only one `id` into this array, when all - set all todos ids into this array. Use it for all actions